presto-common/src/main/java/com/facebook/presto/common/type/EnumTypeConstraint.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.type;
+
+public enum EnumTypeConstraint
+{
+    NONE,
```

Comment:
What does `NONE` mean?

Reply:
`NONE` means no enum type constraint.
The reason I'm not using `null` here is because I wanted to add an enumTypeConstraint parameter with a default value to the @interface `TypeParameter`, but `null` [is not accepted as a default value for @interface parameters](https://stackoverflow.com/questions/1178104/error-setting-a-default-null-value-for-an-annotations-field).

If you think that looks weird, I could make separaate @interfaces instead like `LongEnumTypeParameter` and `VarcharEnumTypeParameter`

Reply:
If you want to add it to TypeParameter, why not add these values to `ParameterKind`?

Reply:
It feels awkward to require every type to have an `EnumTypeConstraint`. I think for TypeParameter it's better to wrap this information into `ParameterKind`. You can keep the `EnumTypeConstraint` in `TypeVariableConstraint` if you need to distinguish between `LongEnumType` and `VarcharEnumType` and use `@Nullable` if you wish.

Reply:
I meant the decorator [TypeParameter](https://github.com/prestodb/presto/blob/67469f2cb0d13ee8825383007bed3e750a67af7a/presto-spi/src/main/java/com/facebook/presto/spi/function/TypeParameter.java#L30), not the [class TypeParameter](https://github.com/prestodb/presto/blob/04138bc26a4fc801630fe936463d89d071d00c0a/presto-common/src/main/java/com/facebook/presto/common/type/TypeParameter.java#L20)

Reply:
That sounds wrong. the `@TypeParameter` works like a generic. This definitely doesn't fit there. If necessary, introduce a new annotation and add it to `ImplementationDependency`.